doc/contributing.rst

```diff
@@ -285,6 +285,22 @@ or
 
 the first is preferred.
 
+Inline mono spaced text can be input using \`single backticks\` or
+\`\`double backticks\`\`. The former are syntactically highlighted,
+the latter are not. To avoid accidental highlighting follow the rule:
+
+* use single backticks for fragments of code in Nim and other
+  programming languages, including identifiers
+
+* prefer double backticks otherwise:
```

Comment:
* we should assume `.. default-role:: nim` only when parsing nim files

* for rst files, we should instead follow what github and rst2html do, assume `.. default-role:: literal` when no role is specified for rst files, that way there are no surprises and code renders in github more similarly to `nim rst2html`; in particular, adding `.. default-role:: code`  would then be the recommended thing to add on top, so it renders as monospace (both in github and nim rst2html), as we btw now already do in $nim/doc/*.rst

`nim rst2html main.rst` seems to be useful beyond pure nim projects so we shouldn't hardcode nim here. Even for nim repos, rst files typically mention other things like compiler output, installation instructions, bash commands, etc, which shouldn't assume nim role but instead the generic literal/code role; this would also avoid need for double backticks in such places.

Reply:
OK, done.

You are probably right, it's better think of nim rst not as an internal Nim tool, but a general one.